pandas/_libs/tslibs/resolution.pyx

```diff
@@ -373,21 +373,27 @@ cdef unique_deltas(ndarray[int64_t] arr):
     return result
 
 
-def _is_multiple(us, mult):
+cdef inline bint _is_multiple(int64_t us, int64_t mult):
     return us % mult == 0
 
 
-def _maybe_add_count(base, count):
+cdef inline str _maybe_add_count(str base, count):
     if count != 1:
         return '{count}{base}'.format(count=int(count), base=base)
     else:
         return base
 
 
-class _FrequencyInferer(object):
+cdef class _FrequencyInferer(object):
     """
     Not sure if I can avoid the state machine here
     """
+    cdef public:
+        index
```

Comment:
Do we need types here?

Reply:
Not really viable since we can't specify `Index` as a type.  We just have to make the name public for wheels to turn.

Reply:
just specify them as object to make this explict